pkgs/development/tools/alloy/default.nix

```diff
@@ -1,58 +1,78 @@
 { stdenv, fetchurl, jre, makeDesktopItem }:
 
-stdenv.mkDerivation rec {
-  name = "alloy-${version}";
-  version = "4.2_2015-02-22";
+let generic = { major, version, src }:
 
-  src = fetchurl {
-    sha256 = "0p93v8jwx9prijpikkgmfdzb9qn8ljmvga5d9wvrkxddccjx9k28";
-    url = "http://alloy.mit.edu/alloy/downloads/alloy${version}.jar";
-  };
-
-  desktopItem = makeDesktopItem rec {
-    name = "alloy";
-    exec = name;
-    icon = name;
-    desktopName = "Alloy";
-    genericName = "Relational modelling tool";
-    comment = meta.description;
-    categories = "Development;IDE;Education;";
-  };
+  stdenv.mkDerivation rec {
+    name = "${nameMajor}-${version}";
+    nameMajor = "alloy${major}";
 
-  buildInputs = [ jre ];
+    desktopItem = makeDesktopItem rec {
+      name = "${nameMajor}";
+      exec = name;
+      icon = name;
+      desktopName = "Alloy ${major}";
+      genericName = "Relational modelling tool";
+      comment = meta.description;
+      categories = "Development;IDE;Education;";
+    };
 
-  phases = [ "installPhase" ];
+    buildInputs = [ jre ];
```

Comment:
Not really used for building. In the wrapper the full path is specified already.